contracts/Vault.vy

```diff
@@ -581,6 +581,14 @@ def setWithdrawalQueue(queue: address[MAXIMUM_STRATEGIES]):
         if queue[i] == ZERO_ADDRESS and self.withdrawalQueue[i] == ZERO_ADDRESS:
             break
         assert self.strategies[queue[i]].activation > 0
+
+        for j in range(MAXIMUM_STRATEGIES):
+            if queue[j] == ZERO_ADDRESS:
+                break
+            if i == j:
+                continue
+            assert queue[i] != queue[j]
```

Comment:
you might be able to get away with `for j in range(min(i, MAXIMUM_STRATEGIES)): ...` and avoid the redundant checks, or simple change the `if i == j: ...` condition to break instead of skip that loop

Reply:
I have tried to do that with `for j in range(i - 1):` but that wasn't compiling

Reply:
Vyper version: 0.2.11
CompilerError: vyper returned the following errors:
```
Value must be a literal
  contract "contracts/Vault.vy", function "setWithdrawalQueue", line 585:8
       584
  ---> 585         for j in range(min(i, MAXIMUM_STRATEGIES)):
  -----------------^
       586             if queue[j] == ZERO_ADDRESS:
```

Reply:
Ah my bad. That was discussed, but never implemented. What I was thinking of was actually this:
```python
    for j in range(i, i + MAXIMUM_STRATEGIES):
        ...
```

Which... doesn't actually help us much in this instance (it allows "chunking up" a large iteration that might take multiple calls to fully work through without breaking the gas limit)

---

I think this would actually be the best:

```python
        # Make sure that `queue[i]` is not duplicated in the list
        for j in range(MAXIMUM_STRATEGIES):
            # NOTE: This will only check entries in `queue` that are `< i` (saves gas)
            if i == j:
                break
            assert queue[i] != queue[j]
```
Note that I removed the breaking condition  `queue[j] == ZERO_ADDRESS` because it shouldn't be possible to find a `j < i` that is an empty address given we already break out above if `queue[i]` is an empty address